### ISSUE TITLE
Breaks if called concurrently more than twice

### ISSUE THREAD

**Issue by amrhassan** [zalando-stups/zign#55]
Calling the script more than twice concurrently breaks the credential caching behaviour.

To reproduce:
```
ztoken & ; ztoken & ; ztoken &
```

I've come across this by trying to call `ztoken` from my concurrent service locally. I'm thinking this could be fixed by a having the script use a lock file to prevent itself from running more than once concurrently.

**Comment by gargravarr** [comment]
Hi @amrhassan . `ztoken`'s purpose is to be used as a command line tool by a user, and given that the user must be authenticated - a browser window is opened if it's not -, I don't see any use case of using it concurrently or in a service.

**Comment by amrhassan** [comment]
Hi @gargravarr. We're using it to generate tokens during local development, and for the sake of minimizing friction we have our services fork this process and capture its stdout. As it happens our service is doing this in a concurrent fashion.

I've created a PR with a fix already and I think it's good in general for executables to be either concurrency safe or take appropriate measures to prevent itself from unexpected breakage :)

**Comment by gargravarr** [comment]
You won't fix any _credential caching behaviour_, because calling `ztoken` with no parameters won't cache the token. You will get a new token when calling it again.

What you can do is use a named token, which returns the same token if already created (and not expired), as in `ztoken token -n mytoken`. And since the token is valid for any forked process, it should be anyway executed before the fork and only once, so I still don't see a valid enough reason to make it concurrent safe :smirk:

**Comment by amrhassan** [comment]
That works for me. Thanks :grin:
